### PR TITLE
feat:session client wrapper

### PR DESCRIPTION
**Pull request by livingrockrises**
## PR-Codex overview
This PR focuses on refining the session management and validation modules within the SDK, enhancing the functionality for session key signers, and improving the structure and readability of the code.

### Detailed summary
- Removed `createSessionNexusClient.ts` file.
- Updated `Types.ts` to include new types and clean up imports.
- Refactored `useSession` decorator for cleaner import structure.
- Added tests for session key functionality in `createNexusSessionClient`.
- Improved error handling and validation in session management.
- Enhanced readability by restructuring logic and comments across multiple files.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/sdk/clients/createSessionNexusClient.ts
Comment (VGabriel45) [inline src/sdk/clients/createSessionNexusClient.ts:1]: Can we rename to "createNexusSessionClient" ?

Comment (livingrockrises) [reply src/sdk/clients/createSessionNexusClient.ts:1]: sure

**Comment by livingrockrises** [comment]
lil unsure about lint changes

File: src/sdk/clients/createNexusSessionClient.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            | ((parameters: {
+                account: account | SmartAccount
+                bundlerClient: Client
+                userOperation: UserOperationRequest
+              }) => Promise<EstimateFeesPerGasReturnType<"eip1559">>)
+            | undefined
+        }
+      | undefined
+    /** Owner of the account. */
+    // Note: Made the signer optional. This can be a session signer?
```
Comment (livingrockrises) [inline src/sdk/clients/createNexusSessionClient.ts:94]: review

**Review by livingrockrises — COMMENTED** [review]
review i

File: src/sdk/clients/createNexusSessionClient.ts (hunk 1 above)
Comment (VGabriel45) [reply src/sdk/clients/createNexusSessionClient.ts:94]: Yes, we can pass here a session signer but I can see that in "createNexusSessionClient we are already building a signer "incompatibleSigner", isn't that the same ?

File: src/sdk/clients/createNexusSessionClient.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    signer?: UnknownSigner
+    /** Index of the account. */
+    index?: bigint
+    /** Active module of the account. */
+    activeModule?: ToValidationModuleReturnType
+    /** Factory address of the account. */
+    factoryAddress?: Address
+
+    // TODO: review below
+    // Note: I added these as things required to send a session tx.
```
[the remark this answers is not in the document]
Comment (VGabriel45) [reply src/sdk/clients/createNexusSessionClient.ts:104]: This is ok. accountName and accountKey can be removed

**Review by VGabriel45 — APPROVED** [review]

**Pull request merged**